src/autowiring/AnySharedPointer.h

```diff
@@ -68,6 +68,8 @@ struct AnySharedPointer {
     return obj && try_assign(obj);
   }
 
+  const std::shared_ptr<void>& as_void(void) const { return m_ptr; }
+
```

Comment:
Also when I use this type I get burned almost every time and have to rely pretty heavily on unit tests to ensure correctness.  It's definitely not the friendliest thing in the world.